File: tests/fixtures/sample.kt
// Sample Kotlin module for Tree-sitter testing.

fun greet(name: String): String {
    return "Hello, $name!"
}

fun farewell(name: String): String {
    return "Goodbye, $name!"
}

class Calculator {
    private val history = mutableListOf<Int>()

    fun add(a: Int, b: Int): Int {
        val result = a + b
        history.add(result)
        return result
    }

    fun multiply(a: Int, b: Int): Int {
        return a * b
    }
}

fun main() {
    val msg = greet("world")
    val bye = farewell("world")
    val calc = Calculator()
    val result = calc.add(2, 3)
    val product = calc.multiply(result, 2)
    println(msg)
}

File: tests/fixtures/sample.rs
/// Sample Rust module for Tree-sitter testing.

fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

fn farewell(name: &str) -> String {
    format!("Goodbye, {}!", name)
}

struct Calculator {
    history: Vec<i64>,
}

impl Calculator {
    fn new() -> Self {
        Calculator { history: Vec::new() }
    }

    fn add(&mut self, a: i64, b: i64) -> i64 {
        let result = a + b;
        self.history.push(result);
        result
    }

    fn multiply(&self, a: i64, b: i64) -> i64 {
        a * b
    }
}

fn main() {
    let msg = greet("world");
    let bye = farewell("world");
    let mut calc = Calculator::new();
    let result = calc.add(2, 3);
    let product = calc.multiply(result, 2);
    println!("{}", msg);
}
